File: docs/workouts/templates/web/workoutDomain.ts
// Reference scaffold only. Keep domain types free of React and Firebase imports.

export type WorkoutLevel = "beginner" | "intermediate" | "advanced";
export type WorkoutVisibility = "private" | "tribe" | "public";

export interface ExerciseAssetManifest {
  lottiePath: string;
  thumbnailPath: string;
  muscleMapFrontPath: string;
  muscleMapBackPath: string;
  assetVersion: number;
  assetHash: string;
}

export interface Exercise {
  id: string;
  name: string;
  primaryMuscles: string[];
  secondaryMuscles: string[];
  equipment: string[];
  level: WorkoutLevel;
  instructions: string[];
  formCues: string[];
  commonMistakes: string[];
  assetManifest: ExerciseAssetManifest;
}

export interface WorkoutCatalogFilter {
  searchText?: string;
  muscleGroup?: string;
  equipment?: string;
  level?: WorkoutLevel;
}

export interface WorkoutCatalogRepository {
  listExercises(filter: WorkoutCatalogFilter): Promise<Exercise[]>;
}

export class ListExercisesUseCase {
  constructor(private readonly repository: WorkoutCatalogRepository) {}

  execute(filter: WorkoutCatalogFilter): Promise<Exercise[]> {
    return this.repository.listExercises(filter);
  }
}

File: docs/workouts/templates/android/WorkoutDomain.kt
package app.tribelog.workouts.template

// Reference scaffold only. Keep domain models free of Compose and Firebase imports.

enum class WorkoutLevel {
    Beginner,
    Intermediate,
    Advanced
}

enum class WorkoutVisibility {
    Private,
    Tribe,
    Public
}

data class ExerciseAssetManifest(
    val lottiePath: String,
    val thumbnailPath: String,
    val muscleMapFrontPath: String,
    val muscleMapBackPath: String,
    val assetVersion: Int,
    val assetHash: String
)

data class Exercise(
    val id: String,
    val name: String,
    val primaryMuscles: List<String>,
    val secondaryMuscles: List<String>,
    val equipment: List<String>,
    val level: WorkoutLevel,
    val instructions: List<String>,
    val formCues: List<String>,
    val commonMistakes: List<String>,
    val assetManifest: ExerciseAssetManifest
)

data class WorkoutCatalogFilter(
    val searchText: String = "",
    val muscleGroup: String? = null,
    val equipment: String? = null,
    val level: WorkoutLevel? = null
)

interface WorkoutCatalogRepository {
    suspend fun listExercises(filter: WorkoutCatalogFilter): List<Exercise>
}

class ListExercisesUseCase(
    private val repository: WorkoutCatalogRepository
) {
    suspend fun execute(filter: WorkoutCatalogFilter): List<Exercise> {
        return repository.listExercises(filter)
    }
}

